src/main/java/com/blackwaterpragmatic/webservices/mybatis/mapper/UserMapper.java

```diff
@@ -1,14 +1,17 @@
 package com.blackwaterpragmatic.webservices.mybatis.mapper;
 
 import com.blackwaterpragmatic.webservices.bean.User;
-
 import org.apache.ibatis.annotations.Param;
 
 import java.util.List;
 
 public interface UserMapper {
 
-	List<User> list();
+	List<User> list(@Param("name") String name);
```

Comment:
The name is optional, but if provided the UserMapper will filter for names containing the given param